plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTableBuilder.java

```diff
@@ -64,53 +68,62 @@ public static DynamicTable buildFromPql(PinotMetadata pinotMetadata, SchemaTable
+                ExpressionContext expressionContext = orderByExpressionContext.getExpression();
+                checkState(expressionContext.getType() == ExpressionContext.Type.IDENTIFIER, "Unexpected order by expression: '%s'", expressionContext);
+                String name = expressionContext.getIdentifier();
+                PinotColumnHandle columnHandle = getColumnHandle(schemaTableName, name, columnHandles);
+                orderByBuilder.add(new OrderByExpression(columnHandle.getColumnName(), orderByExpressionContext.isAsc()));
             }
+            orderBy = orderByBuilder.build();
         }
 
-        List<String> groupByColumns;
-        if (request.getGroupBy() == null) {
-            groupByColumns = ImmutableList.of();
-        }
-        else {
-            groupByColumns = resolvePinotColumns(schemaTableName, request.getGroupBy().getExpressions(), columnHandles);
+        Optional<String> filter = Optional.empty();
+        if (pinotQuery.getFilterExpression() != null) {
+            String formatted = formatFilter(schemaTableName, queryContext.getFilter(), columnHandles);
+            filter = Optional.of(formatted);
         }
 
-        Optional<String> filter;
-        if (request.getFilterQuery() != null) {
-            filter = Optional.of(convertFilter(request.getPinotQuery(), columnHandles));
-        }
-        else {
-            filter = Optional.empty();
-        }
-        QueryContext queryContext = BrokerRequestToQueryContextConverter.convert(request);
+        List<String> selectionColumns = ImmutableList.of();
+        List<String> groupByColumns = ImmutableList.of();
+
         ImmutableList.Builder<PinotColumnHandle> aggregateColumnsBuilder = ImmutableList.builder();
-        if (request.getAggregationsInfo() != null) {
+        if (queryContext.getAggregationFunctions() != null) {
+            checkState(queryContext.getAggregationFunctions().length > 0, "Aggregation Functions is empty");
             for (AggregationFunction aggregationFunction : queryContext.getAggregationFunctions()) {
                 aggregateColumnsBuilder.add(new PinotColumnHandle(
                         aggregationFunction.getResultColumnName(),
                         toTrinoType(aggregationFunction.getFinalResultColumnType())));
             }
+            if (queryContext.getGroupByExpressions() != null) {
+                groupByColumns = getPinotColumnNames(schemaTableName, queryContext.getGroupByExpressions(), columnHandles);
+            }
+        }
+        else {
+            checkState(queryContext.getSelectExpressions() != null && !queryContext.getSelectExpressions().isEmpty(), "Pinot selections is null or empty");
+            checkState(queryContext.getAliasList() == null || queryContext.getAliasList().isEmpty() || !queryContext.getAliasList().stream().anyMatch(alias -> alias != null), "Aliases are not supported");
```

Comment:
Seems like we couldn't check this earlier but do now. Correct?

Reply:
Yes, the new QueryContext has an alias list which we did not use before (pinot is deprecating/eliminating broker request "pql" and moving it all to QueryContext). I've also added support for aliases but will put it in a subsequent pr as it changes functionality.